Fixes crontab throwing errors when tmux isn't running

AmirGhiassian opened the pull request:
Previously, when the plugin was installed, changing directories or every couple minutes it would throw an error because there wasn't any tmux session. So I did this:

If you run `echo $TERM_PROGRAM` in tmux, the value will be `tmux`

So first commit I did was to test this fix to see if it works on one function, so then I implemented it in a function and put them for everything: 

```bash
# Check if we are in tmux
check_tmux() {
  if [ "$TERM_PROGRAM" = "tmux" ]; then
    true
  else
    false
  fi
}
```

_That's all! Tested on my Mac-book M1 Pro, running Mac OS V15.0 beta_

AmirGhiassian committed:
Added check to see if tmux session is open

And formatted code a little bit. Reason for this check is because the
cron job runs even if tmux isn't open and causes error messages because
there isn't any active tmux session

AmirGhiassian committed:
Added function to check for tmux & added to 2 functions

AmirGhiassian commented:
_Edit: My prettier plugin also formatted the lines so that is why the lines changed is high_

AmirGhiassian commented:
Never mind, didn't fix the issue:
<img width="625" alt="image" src="https://github.com/user-attachments/assets/47dd0d30-4430-4053-8130-ee60c90f3fcc">

thepante commented:
Hey thanks for the heads up.
That's right, I saw that error on the shell when not under tmux. That's something to address!

As I understood, the capture should be about something else.
That seems to be the job daemon log. It should be suppressed by editing the cronjob:
```sh
crontab -e
```
And there append this `> /dev/null 2>&1` to this plugin job, something like this and save it:
```sh
*/1 * * * * /Users/thepante/.tmux/plugins/tmux-git-autofetch/git-autofetch.tmux --scan-paths > /dev/null 2>&1
```

That should suppress that noise under mail.
Let me know if that worked.

I'll better check standards about how to handle cronjobs and update it 😅

---

Will check out your PR and merge it! Thanks for it.

AmirGhiassian commented:
Hi, sorry for the late response.

Thank you for the temp fix, I won't apply it for myself just yet, il try and see if I can help out and adjust the cron job 😁

AmirGhiassian commented:
Don't merge yet, I messed up my gpg key and also I haven't tested it yet.

AmirGhiassian committed:
Simplified checking

Instead of checking in each function, it will check before the switch
statement, less lines of code = efficient code =)

Signed-off-by: Amirreza Ghiassian <amir@Amirs-Macbook.local>

AmirGhiassian committed:
Changed it to this for if statement

I may be bad at bash programming

AmirGhiassian committed:
Removed old fix b/c not working

AmirGhiassian committed:
Fixed crontab entry

AmirGhiassian commented:
Should be fixed now 👍🏽

AmirGhiassian commented:
uhhhh, bump?

AmirGhiassian commented:
b-bump?

thepante commented:
@AmirGhiassian done, merged! Thank you so much for the contribution!
sorry for the delay tho 😔

AmirGhiassian commented:
No worries @thepante! I'm glad i could contribute :))
